src/Swarm/Game/Entity.hs

```diff
@@ -604,6 +605,16 @@ deleteAll e (Inventory cs byN h) =
  where
   n = (fst <$> IM.lookup (e ^. entityHash) cs) ? 0
 
+-- | Delete entry for a certain entity from an inventory.
+deleteEntry :: Entity -> Inventory -> Inventory
+deleteEntry e (Inventory cs byN h) =
+  Inventory
+    (IM.delete (e ^. entityHash) cs)
+    (M.delete (T.toLower $ e ^. entityName) byN)
+    (h - (n + 1) * (e ^. entityHash))
```

Comment:
This formula is guesswork.  Can I actually just replace the whole line with `0`?